hxd/SceneEvents.hx

```diff
@@ -200,11 +229,21 @@ class SceneEvents {
 		if( cancelFocus && currentFocus != null )
 			blur();
 
-		if( checkOver && currentOver != null ) {
-			onOut.cancel = false;
-			currentOver.handleEvent(onOut);
-			if( !onOut.cancel )
-				currentOver = null;
+		if( checkOver ) {
+			if ( overIndex < overList.length ) {
+				do {
+					onOut.cancel = false;
+					overList[overIndex].handleEvent(onOut);
+					if ( !onOut.cancel ) {
+						overList.remove(overList[overIndex]);
+						continue;
+					}
+					overIndex++;
+				} while ( overIndex < overList.length );
+				if ( overList.length > 0 ) {
+					hxd.System.setCursor(overList[0].cursor);
+				}

```

Comment:
Two things here:
- I think it's more appropriate to dispatch Out events in the reverse order the Over were dispatched. So since we dispatch Child Over, Parent Over, ParentParent Over, we should dispatch ParentParent Out, ParentOut, Child Out, so iterate in reverse order here
- wha if overList.length == 0 ? We should reset the cursor to Default then

Reply:
Minors styling too : please merge the if conditions, such as `if( checkOver && overIndex < overList.length )`

Reply:
When `overList.length == 0` cursor is being reset to Default by `Interactive.handleEvent` .
See here:
https://github.com/HeapsIO/heaps/blob/d1bff99b8f43f2a362fc60af846fcd3637b04d1e/h2d/Interactive.hx#L179-L183
and here:
https://github.com/HeapsIO/heaps/blob/d1bff99b8f43f2a362fc60af846fcd3637b04d1e/h3d/scene/Interactive.hx#L99-L103

Reply:
I think while we're at it we should let the SceneEvents class handle cursor changes entirely. Actually there's a bugfix to be done as now setting cursor will update it if isOver() while it should only do if we're the "first" (topmost) over with a defined cursor. Maybe calling SceneEvents.cursorUpdated(this) is better

Reply:
We might also want to minimize cursor changes by only calling setCursor once after all Over's done.

Reply:
In case of `cursor == null`, should it fall trough to the next Interactive in the over list or use `Default`?

Reply:
I think it should fall through

Reply:
It's still calls `SceneEvents.selectCursor` when cursor is changed on non-top overed cursor, but this is due to fall trough logic, because there may be a case where first Interactive will have `cursor == null` and updated cursor should actually take place.

Reply:
Makes sense, will review now